src/main/java/DukeCommand.java

```diff
@@ -0,0 +1,203 @@
+
+                LocalDate date = LocalDate.parse(inputdate[0],
+                        formatdate.withResolverStyle(ResolverStyle.STRICT));
+
+                // Format it to english. For example, 3 Oct 2019
+                inputdate[0] = date.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+
+
+                // Time Validation
+                if ((inputdate[1].toCharArray()).length != 4)
+                    throw new DukeException("Please enter time in the form of 24 hour format!");
+                int dateInt = Integer.parseInt(inputdate[1]);
+                int timeTest = dateInt % 100;
+                // Check if the minute is valid
+                if ((timeTest > 60 || timeTest < 0) || (dateInt > 2359 || dateInt < 0))
+                    throw new DukeException("Invalid time!");
+
+                String time = "";
+
+                // Translating the time from 24-hour format to
+                // AM/PM format
+                if (dateInt < 1300){
+                    time = timeTest < 10 ? time + "0" : time;
+
+                    String hour = dateInt < 100 ? "12" : Integer.toString(dateInt / 100);
+                    String minute = Integer.toString(timeTest);
+                    time = dateInt < 1200 ? hour + ":" + time + minute + "am" :
+                            hour + ":" + time + minute + "pm";
+                } else {
+                    time = (dateInt % 100) < 10 ? time + "0" : time;
+                    String hour = Integer.toString((dateInt / 100) - 12);
+                    String minute = Integer.toString(timeTest);
+                    time = hour + ":" + time + minute + "pm";
+                }
+                inputdate[1] = time;
+
+                String newstr = String.join(" ", inputdate);
+                list.add(new Event(s1.substring(6, limit), newstr));
+
+            } catch (DateTimeParseException exception){
+                ui.showErrorMessage(exception.getMessage() + "\n" +
+                        "Input date in the form of yyy-mm-dd and " +
+                        "remember to add time in 24-hour format");
+            } catch (Exception ex) {
+                ui.showErrorMessage(ex.getMessage());
+            }
+
+        }
+    },
+    DELETE {
+        @Override
+        public void execute(String s1, TaskList list, Ui ui, Storage storage) {
+            // Split the string to get the
+            // index of the task to be deleted
+            String[] arrdel = s1.split("\\s+");
+            int pos = Integer.parseInt(arrdel[1]) - 1;
+
+            list.delete(pos, storage);
+        }
+    },
+    FIND {
+      @Override
+      public void execute(String s1, TaskList list, Ui ui, Storage storage) {
+          String[] commandarray = s1.split("\\s+", 2);
+          String keyword = commandarray[1];
+
+          list.find(keyword, storage);
+      }
+    };
+
+    public abstract void execute(String s1, TaskList list, Ui ui, Storage storage);
```

Comment:
This is a very good use of enums! It makes the code for executing Duke commands much more elegant and readable. If time allows, I would like to adopt this structure in my own Duke Project as well XD